src/main/java/de/tum/in/www1/artemis/web/rest/LectureResource.java

```diff
@@ -383,4 +386,15 @@ public ResponseEntity<Void> deleteLecture(@PathVariable Long lectureId) {
         lectureService.delete(lecture);
         return ResponseEntity.ok().headers(HeaderUtil.createEntityDeletionAlert(applicationName, true, ENTITY_NAME, lectureId.toString())).build();
     }
+
+    /**
+     * This method calls the executeLectureIngestionPipeline function with the needed parameters in the right format
+     *
+     * @param lecture the lecture to be updated
+     */
+    private void helpExecuteIngestionPipeline(Lecture lecture) {
+        List<AttachmentUnit> attachmentUnitList = lecture.getLectureUnits().stream().filter(lectureUnit -> lectureUnit.getType().equals("attachment"))
+                .map(lectureUnit -> (AttachmentUnit) lectureUnit).collect(Collectors.toCollection(ArrayList::new));
+        pyrisWebhookService.ifPresent(service -> service.executeLectureIngestionPipeline(true, attachmentUnitList));
```

Comment:
Doing the filtering inside the lambda block only when the PyrisWebhookService is present can make this endpoint more performant.
```suggestion
        pyrisWebhookService.ifPresent(service -> {
                    List<AttachmentUnit> attachmentUnitList = lecture.getLectureUnits().stream().filter(lectureUnit -> lectureUnit.getType().equals("attachment"))
                .map(lectureUnit -> (AttachmentUnit) lectureUnit).collect(Collectors.toCollection(ArrayList::new));
            service.executeLectureIngestionPipeline(true, attachmentUnitList)
        });
```